### PR TITLE
ISPN-13857 Update Caffeine to 2.9.3

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-13857

### CODE REVIEW

File: core/src/main/java/org/infinispan/container/impl/BoundedSegmentedDataContainer.java [hunk 1]
```diff
@@ -103,13 +96,35 @@ protected void computeEntryWritten(K key, InternalCacheEntry<K, V> value) {
...
+   protected void computeEntryRemoved(int segment, K key, InternalCacheEntry<K, V> value) {
       ConcurrentMap<K, InternalCacheEntry<K, V>> map = BoundedSegmentedDataContainer.super.getMapForSegment(segment);
       if (map != null) {
          map.remove(key, value);
       }
    }
 
+   @Override
+   protected void putEntryInMap(PeekableTouchableMap<K, V> map, int segment, K key, InternalCacheEntry<K, V> ice) {
+      map.computeIfAbsent(key, k -> {
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/container/impl/BoundedSegmentedDataContainer.java:111]: is it intentional to use `computeIfAbsent`?

**Review by pruivo — COMMENTED** [review]
LGTM. Just have a small question.

**Comment by pruivo** [comment]
CI is failing badly (100 failures)

**Comment by wburns** [comment]
> CI is failing badly (100 failures). related?

They are all hibernate related... which is odd. But I will look into them.

**Comment by wburns** [comment]
> 

Wasn't sure how it was possible, but I rebased and now it is even worse, haha.

**Comment by pruivo** [comment]
that's my fault :rofl: I merged Don's change without looking at CI (I'm used to that docs do not mess up CI)

File: core/src/main/java/org/infinispan/container/impl/BoundedSegmentedDataContainer.java (hunk 1 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/container/impl/BoundedSegmentedDataContainer.java]: No, should really be `compute`.

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @wburns !

**Pull request merged**